Sources/StreamVideoSwiftUI/Utils/PictureInPicture/StreamBufferTransformer.swift

```diff
@@ -0,0 +1,214 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import StreamVideo
+import StreamWebRTC
+
+/// `StreamBufferTransformer` is a struct that provides methods for transforming RTCI420Buffer to
+/// CVPixelBuffer. while performing downsampling when necessary.
+struct StreamBufferTransformer {
+
+    var requiresResize = false
+
+    /// Transforms an RTCI420Buffer to a CVPixelBuffer with optional resizing.
+    ///
+    /// - Parameters:
+    ///   - source: The source RTCI420Buffer to be transformed.
+    ///   - targetSize: The target size for the resulting CVPixelBuffer.
+    /// - Returns: A transformed CVPixelBuffer or nil if transformation fails.
+    func transform(
+        _ source: RTCI420Buffer,
+        targetSize: CGSize
+    ) -> CVPixelBuffer? {
+        let sourceSize = CGSize(width: Int(source.width), height: Int(source.height))
+
+        guard
+            requiresResize,
+            let resizedSource = resize(source, to: resizeSize(sourceSize, toFitWithin: targetSize))
```

Comment:
Resizing using WebRTC's downscaling methods is actually more performant than rendering a very big (in dimensions) RTCVideoFrame. With this in place we can render tracks irregardless of their size without skyrocketing performance (that applies to both screensharing or not).

Reply:
nice find 👍